pkg/webhook/controlplane/ensurer.go

```diff
@@ -40,13 +44,33 @@ func (e *ensurer) InjectClient(client client.Client) error {
 
 // EnsureKubeAPIServerDeployment ensures that the kube-apiserver deployment conforms to the provider requirements.
 func (e *ensurer) EnsureKubeAPIServerDeployment(ctx context.Context, ectx genericmutator.EnsurerContext, new, _ *appsv1.Deployment) error {
+	cluster, _ := ectx.GetCluster(ctx)
+	makeAuditForwarder := false
+
+	if e.controllerConfig.ClusterAudit.Enabled {
+		cpConfig, err := helper.ControlPlaneConfigFromClusterShootSpec(cluster)
+		if err != nil {
+			logger.Error(err, "Could not read ControlPlaneConfig from cluster shoot spec", "Cluster name", cluster.ObjectMeta.Name)
```

Comment:
Can we return an error here such that the webhook fails?

```suggestion
			logger.Error(err, "Could not read ControlPlaneConfig from cluster shoot spec", "Cluster name", cluster.ObjectMeta.Name)
			return err
```

Reply:
Should we though?
If we return immediately, the authn webhook does not get deployed either, and the cluster is more broken than it needs to be.

Reply:
I guess we should because otherwise it looks in Gardener like everything is working when actually it's not. Errors in the reconciliation should be reflected in the overall shoot health.